test/test_encryption.py

```diff
@@ -1945,28 +1946,38 @@ def test_via_loading_shared_library(self):
     # https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#20-bypass-creating-mongocryptd-client-when-shared-library-is-loaded
     @unittest.skipUnless(os.environ.get("TEST_CRYPT_SHARED"), "crypt_shared lib is not installed")
     def test_client_via_loading_shared_library(self):
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
-            key_vault = client_context.client.keyvault.datakeys
-            key_vault.drop()
-            key_vault.create_index(
-                "keyAltNames",
-                unique=True,
-                partialFilterExpression={"keyAltNames": {"$exists": True}},
-            )
-            key_vault.insert_one(json_data("external", "external-key.json"))
-            schemas = {"db.coll": json_data("external", "external-schema.json")}
-            opts = AutoEncryptionOpts(
-                kms_providers={"local": {"key": LOCAL_MASTER_KEY}},
-                key_vault_namespace="keyvault.datakeys",
-                schema_map=schemas,
-                mongocryptd_uri="mongodb://localhost:27021",
-                crypt_shared_lib_required=True,
-            )
-            client_encrypted = rs_or_single_client(auto_encryption_opts=opts)
-            self.addCleanup(client_encrypted.close)
-            client_encrypted.db.coll.drop()
-            client_encrypted.db.coll.insert_one({"encrypted": "test"})
-            sock.bind(("localhost", 27021))
+        def listen():
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                addr = ("localhost", 27021)
+                sock.bind(addr)
+                sock.settimeout(2)
```

Comment:
This is technically a race condition.  If we instead add a non-blocking poll on accept, and raise an error if we do accept, but use a queue to pass a "stop" sentinel into the thread, we can remove the race condition.

Reply:
I would also prefer to refactor this so that the test doesn't take 2 seconds. Ideally it should run as fast as possible.

Reply:
We can probably use https://docs.python.org/3/library/socketserver.html for this to  avoid manually writing the setup and shutdown logic.